File: 728-self-dividing-numbers/self-dividing-numbers.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {

        ArrayList<Integer> list = new ArrayList<Integer>();

        for(int i = left ; i <= right ; i++){
            int temp = i;
            int flag = 0;

            while(temp > 0){
                int d = temp%10;

                if( d == 0){
                    break;
                }
                if(i%d == 0){
                    flag += 1;
                }else{
                    break;
                }
                temp /= 10;
            }
            if(flag == String.valueOf(Math.abs(i)).length()){
                list.add(i);
            }
        }
        return list;
    }
}

File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> list[int]:
        ans = []
        for i in range(left,right+1):
            temp = i
            flag = 0
            while temp > 0:
                d = temp%10
                if d == 0:
                    break
                if i%d == 0:
                    flag += 1
                else:
                    break

                temp //= 10             
            if flag == len(str(i)):
                ans.append(i)
        return ans        
